hydra_base/lib/users.py

```diff
@@ -281,12 +284,14 @@ def reset_failed_logins(username, flush=True, **kwargs):
     try:
         user_i = db.DBSession.query(User).filter(User.username == username).one()
     except NoResultFound:
-        raise HydraError(username)
+        # raise HydraError(username)
+        raise HydraLoginUserNotFound("User '{}' does not exist".format(username))
 
     user_i.failed_logins = 0
 
     if flush is True:
         db.DBSession.flush()
+        transaction.commit()
```

Comment:
See previous comments.

Reply:
Without the commit, a following flush got locked forever and breaks the code flow with a db exception on the max-lock-timeout exceeded.

Reply:
Hmmm. I wonder is this then not a part of the scoped session workflow? This should be taken care of by zope.
Ah wait. Yes, i see. Question is: where is the this login called from? HWI?

Reply:
Yes it is called from HWI

Reply:
has this been removed?